powersimdata/design/generation/clean_capacity_scaling.py

```diff
@@ -174,24 +174,35 @@ def add_resource_data_to_targets(input_targets, scenario, calculate_curtailment=
     capacity_groupby = plant.Pmax.groupby(groupby_cols)
     capacity_by_target_type = capacity_groupby.sum().unstack(fill_value=0)
     # Generated energy
-    pg_groupby = scenario.state.get_pg().sum().groupby(groupby_cols)
+    pg_groupby = scenario.get_pg().sum().groupby(groupby_cols)
     summed_generation = pg_groupby.sum().unstack(fill_value=0)
     # Calculate capacity factors
     possible_energy = scenario_length * capacity_by_target_type[curtailment_types]
     capacity_factor = summed_generation[curtailment_types] / possible_energy
+
     if calculate_curtailment:
         # Calculate: curtailment, no_curtailment_cap_factor
         # Hydro and solar are straightforward
-        hydro_plant_sum = scenario.state.get_hydro().sum()
-        hydro_plant_targets = plant[plant.type == "hydro"].target_area
+        hydro_plant_sum = scenario.get_profile("hydro").sum()
+        hydro_plant_targets = plant[
+            plant["type"].isin(
+                grid.model_immutables.plants["group_profile_resources"]["hydro"]
+            )
+        ]["target_area"]
         hydro_potential_by_target = hydro_plant_sum.groupby(hydro_plant_targets).sum()
-        solar_plant_sum = scenario.state.get_solar().sum()
-        solar_plant_targets = plant[plant.type == "solar"].target_area
+        solar_plant_sum = scenario.get_profile("solar").sum()
+        solar_plant_targets = plant[
+            plant["type"].isin(
+                grid.model_immutables.plants["group_profile_resources"]["solar"]
+            )
+        ]["target_area"]
         solar_potential_by_target = solar_plant_sum.groupby(solar_plant_targets).sum()
-        # Wind is a little tricker because get_wind() returns 'wind' and 'wind_offshore'
-        onshore_wind_plants = plant[plant.type == "wind"].index
-        onshore_wind_plant_sum = scenario.state.get_wind().sum()[onshore_wind_plants]
-        wind_plant_targets = plant[plant.type == "wind"].target_area
+        onshore_wind_type = {
+            v: k for k, v in grid.model_immutables.plants["type2label"].items()
+        }["Onshore Wind"]
```

Comment:
I'm trying to understand the logic here, we do have "label2type" returned in `get_plants` function which is populated into `model_immutables.plants`, right? Also, we have "Onshore Wind" label only for Europe grid, no?

Reply:
You are right that we have `label2type` and there is no need to revert the dict. For some reason, I believed I renamed the label for USA but it seems I did not. It should be:
```
    def __init__(self):
        self.type2label = {
            "nuclear": "Nuclear",
            "geothermal": "Geothermal",
            "coal": "Coal",
            "dfo": "Diesel Fuel Oil",
            "hydro": "Hydro",
            "ng": "Natural Gas",
            "solar": "Solar",
            "wind": "Onshore Wind",
            "wind_offshore": "Offshore Wind",
            "biomass": "Biomass",
            "other": "Other",
            "storage": "Storage",
        }
        self.curtailable2label = {
            "solar": "Solar Curtailment",
            "wind": "Onshore Wind Curtailment",
            "wind_offshore": "Offshore Wind Curtailment",
        }
```

Reply:
Done